### PR TITLE
fix exception reporting in ktest

### LINKED ISSUE #485: ktest throws exception for wrong typed option
```
$ ktest config.xml --update out
org.apache.commons.cli.UnrecognizedOptionException: Unrecognized option: --update
    at org.apache.commons.cli.Parser.processOption(Parser.java:363)
    at org.apache.commons.cli.Parser.parse(Parser.java:199)
    at org.apache.commons.cli.Parser.parse(Parser.java:85)
    at org.kframework.ktest.CmdArgs.CmdArgParser.<init>(CmdArgParser.java:81)
    at org.kframework.ktest.KTest.<init>(KTest.java:36)
    at org.kframework.ktest.KTest.main(KTest.java:107)
    at org.kframework.main.Main.main(Main.java:58)
Exception in thread "main" java.lang.NullPointerException
    at org.kframework.ktest.KTest.main(KTest.java:112)
    at org.kframework.main.Main.main(Main.java:58)
```

### PR DESCRIPTION
**Pull request by osa1**
Fixes #485

### CODE REVIEW

**Comment by osa1** [comment]
Also fixes #519 .

**Comment by osa1** [comment]
Also fixes #523 .

File: src/javasources/KTool/src/org/kframework/ktest/KTest.java [hunk 1]
```diff
@@ -119,6 +119,11 @@ public static boolean main(String[] args) {
                     new KException(KException.ExceptionType.ERROR,
                             KException.KExceptionGroup.CRITICAL,
                             e.getMessage(), location.getSystemId(), location.getPosStr()));
+        } catch (ParseException | InvalidArgumentException e) {
+            GlobalSettings.kem.register(
+                    new KException(KException.ExceptionType.ERROR,
+                            KException.KExceptionGroup.CRITICAL,
+                            e.getMessage(), "command line", "System file."));
```
Comment (dwightguth) [inline src/javasources/KTool/src/org/kframework/ktest/KTest.java]: haha, no. if there's no file or location, omit those parameters, don't put useless junk.

**Comment by dwightguth** [comment]
The correct pattern here is to have a command line option --debug which is a boolean and, when specified, causes exceptions which exit the application to print a stack trace. So, for example:

```
} catch (SAXException e) {
    if (debug) {
        e.printStackTrace();
    }
    GlobalSettings.kem.register(...);
}
```

**Comment by osa1** [comment]
I fixed some irregularities on the way. I suggest reviewing this commit by commit.

**Comment by dwightguth** [comment]
I don't like the way you are handling the case when options parsing fails, but it's not worth fighting over because this is still using the old options framework that I will be getting rid of eventually. I will merge this.

**Comment by osa1** [comment]
> I don't like the way you are handling the case when options parsing fails, but it's not worth fighting over because this is still using the old options framework that I will be getting rid of eventually.

Do you have any suggestions? I have to search through the argument list manually.

**Comment by dwightguth** [comment]
No, as I said, it's fine the way it is now, because we don't really have a
better solution. But when I redo this with JCommander, it will get dealt
with. Don't worry about it.

On Mon, Jun 2, 2014 at 11:04 AM, Ömer Sinan Ağacan <notifications@github.com

**Pull request merged**